File: Data Structures & Algorithms/max-water-container/submission-1.cpp
class Solution {
public:
    int maxArea(vector<int>& heights) {
        int lo = 0;
        int hi = heights.size() - 1;

        int max_area = 0;

        while (lo < hi) {

            int area = (hi - lo) * min(heights[lo], heights[hi]);

            max_area = max(max_area, area);

            if (heights[lo] < heights[hi]) {
                lo++;
            }
            else {
                hi--;
            }
        }

        return max_area;
    }
};

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-1.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        temp = ""
        a = 0

        for c in s:
            if c not in temp:
                temp += c
            else:
                a = max(a, len(temp))
                temp = temp[temp.index(c) + 1:]
                temp += c

        return max(a, len(temp))
        